src/AzureStorage/Queues/AzureStorageQueue.cs

```diff
@@ -153,17 +169,17 @@ public class AzureStorageQueue<T> : QueueBase<T> where T : class {
             _workerErrorCount = 0;
         }
 
-        public override void StartWorking(Func<IQueueEntry<T>, CancellationToken, Task> handler, bool autoComplete = false, CancellationToken cancellationToken = new CancellationToken()) {
+        protected override void StartWorkingImpl(Func<IQueueEntry<T>, CancellationToken, Task> handler, bool autoComplete = false, CancellationToken cancellationToken = new CancellationToken()) {
             if (handler == null)
                 throw new ArgumentNullException(nameof(handler));
-
+            
             var linkedCancellationToken = CancellationTokenSource.CreateLinkedTokenSource(_queueDisposedCancellationTokenSource.Token, cancellationToken).Token;
 
             Task.Run(async () => {
```

Comment:
In the base implementation we were calling Wait and unwrap exceptions? I'm curious on how that will work I think we had deadlocks in the past because of this. Maybe this code should be moved out to the base class.. I think we might need to have another look at this. Maybe start working should just return a Task and we handle iti n the base implementation? But then the handler passed in returns a Task... Can this be simplified?

@ejsmith